### DISCUSSION TITLE
support for the series x controller?

### DISCUSSION THREAD

**Discussion by SoosTieger** [ndeadly/MissionControl#208] [Ideas]
i want to pair the series x controller to the switch is this possible maybe with an update?

**Comment by ndeadly** [comment]
This has been asked several times now. I'm working on it, there's an issue tracking it here https://github.com/ndeadly/MissionControl/issues/135

The problem is that this controller uses Bluetooth LE, rather than BR/EDR like most other controllers. While there is support for BLE in the Switch's bluetooth service, there is no code present in HOS for managing BLE controllers at all, besides some Pokeball Plus specific stuff (which implements it's own custom services, not a standard HID profile). This means that all the code for discovering/pairing, connection/disconnection, input handling etc needs to be added. On top of this, it needs to integrate with the existing controller handling somehow so that these controllers are taken into account when assigning player numbers, disconnecting when the console goes to sleep etc. while also not breaking Pokeball Plus support. I am slowly making progress towards this, but there is still a lot of work required before it's working and stable enough for release.

**Reply by SoosTieger** [reply]
Hit me up when the upate for the series x controller comes...
